### PR TITLE
Feature: upload images with S3

### PR DESCRIPTION
**Pull request by violog**

### CODE REVIEW

File: internal/services/api/handlers/ctx.go [hunk 1]
```diff
@@ -68,3 +70,13 @@ func CtxIssuerConfig(cfg config.IssuerConfig) func(context.Context) context.Cont
 func IssuerConfig(r *http.Request) config.IssuerConfig {
 	return r.Context().Value(issuerConfigKey).(config.IssuerConfig)
 }
+
+func CtxS3(s3 *s3.Storage) func(context.Context) context.Context {
+	return func(ctx context.Context) context.Context {
+		return context.WithValue(ctx, storageCtxKey, s3)
```
Comment (olegfomenko) [inline internal/services/api/handlers/ctx.go:76]: storage ctx key is already in use, add another one

File: internal/services/api/handlers/ctx.go [hunk 2]
```diff
@@ -68,3 +70,13 @@ func CtxIssuerConfig(cfg config.IssuerConfig) func(context.Context) context.Cont
...
+	}
+}
+
+func S3(r *http.Request) *s3.Storage {
+	return r.Context().Value(storageCtxKey).(*s3.Storage)
```
Comment (olegfomenko) [inline internal/services/api/handlers/ctx.go:81]: same

File: internal/services/api/handlers/upload_form.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	form, err := S3(r).UploadFormData(key, request.Attributes.ContentType)
+	if err != nil {
+		Log(r).WithError(err).Error("failed to build form data to upload document")
+		ape.RenderErr(w, problems.InternalError())
+		return
+	}
+
+	attributes := make(map[string]any, len(form))
```
Comment (olegfomenko) [inline internal/services/api/handlers/upload_form.go:37]: why we need this additional map?

File: internal/config/s3.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		result, err := s3.NewMediaTypes(mediaTypes)
+		if err != nil {
+			return reflect.Value{}, errors.Wrap(err, "failed to create new media types")
+		}
+
+		return reflect.ValueOf(result), nil
+	},
+}
+
+func (c *config) S3() *s3.Storage {
```
Comment (olegfomenko) [inline internal/config/s3.go:41]: update config example file

File: internal/data/s3/media_type.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+var mapNameToDocType = map[string]int32{
+	"general": docTypeGeneral,
+	"logo":    docTypeLogo,
+}
+
+type MediaTypes struct {
+	allowed map[int32][]string
+}
+
+func NewMediaTypes(mediaTypes map[string][]string) (MediaTypes, error) {
```
[the remark this answers is not in the document]
Comment (violog) [reply internal/data/s3/media_type.go:23]: I will add this check in handler to make use of it

**Review by olegfomenko — APPROVED** [review]

**Pull request merged**